### PR TITLE
Updated tangential matrix computation

### PR DESCRIPTION
**Pull request by MarkoLeskovar**
Dear Andreas,

I finished with the selected tasks that you gave me. I this branch I did the following:

1. Fixed the GiD input files so that they work with the latest parser and non-linear analysis
2. Relaxed the tolerances in unit tests so that they also pass on Windows machine
3. Implement page-wise computation of tangential stiffness matrix for CST
4. Compared results of new function with the old tanMtx computation and they give the same results

I noticed however, that the solver related to the transient plate in membrane action analysis is missing from the GiD repository, even though "main" script and also an unit test for it exist. Is this intentional or did you somehow deleted it by mistake? 

Please write me an email if you have any questions or remarks.

Best regards,
Marko Leskovar

### CODE REVIEW

File: unitTest/@testFEMComputationalFluidDynamicsAnalysis/testFEM4NavierStokesTransientALE2D.m [hunk 1]
```diff
@@ -36,7 +36,7 @@ function testFEM4NavierStokesTransientALE2D(testCase)
 %% 0. Read input
 
 % Define absolute tolerance
-absTol = 1e-15; % tolerance w.r.t our value
+absTol = 1e-14; % tolerance w.r.t our value
```
Comment (andreas-apostolatos) [inline unitTest/@testFEMComputationalFluidDynamicsAnalysis/testFEM4NavierStokesTransientALE2D.m]: Please revert the changes in the unit tests, I would like to fix them myself under Windows.

File: unitTest/@testFEMComputationalFluidStructureInteractionAnalysis/testFluidStructureInteractionSpringMass.m [hunk 2]
```diff
@@ -36,7 +36,7 @@ function testFluidStructureInteractionSpringMass(testCase)
 %% 0. Read input
 
 % Define absolute tolerance
-absTol = 1e-15;
+absTol = 1e-8;
```
Comment (andreas-apostolatos) [inline unitTest/@testFEMComputationalFluidStructureInteractionAnalysis/testFluidStructureInteractionSpringMass.m]: Please revert the changes in the unit tests, I would like to fix them myself under Windows.

File: unitTest/@testFEMContactMechanicsAnalysis/testFrictionlessSignoriniContactHertz2D.m [hunk 3]
```diff
@@ -38,7 +38,7 @@ function testFrictionlessSignoriniContactHertz2D(testCase)
 
 % Define tolerances for both cases
 absTol = 1e-15;
-absTol5 = absTol*1e5;
+absTol5 = absTol*1e6;
```
Comment (andreas-apostolatos) [inline unitTest/@testFEMContactMechanicsAnalysis/testFrictionlessSignoriniContactHertz2D.m]: Please revert the changes in the unit tests, I would like to fix them myself under Windows.

File: unitTest/@testFEMContactMechanicsAnalysis/testFrictionlessSignoriniContactWedge2D.m [hunk 4]
```diff
@@ -37,7 +37,7 @@ function testFrictionlessSignoriniContactWedge2D(testCase)
 % Define tolerances
 absTol = 1e-15;
 absTol0 = absTol*0;
-absTol2 = absTol*1e2;
+absTol2 = absTol*1e3;
```
Comment (andreas-apostolatos) [inline unitTest/@testFEMContactMechanicsAnalysis/testFrictionlessSignoriniContactWedge2D.m]: Please revert the changes in the unit tests, I would like to fix them myself under Windows.

File: unitTest/@testFEMThermalConductionAnalysis/testTransientSquareCavity.m [hunk 5]
```diff
@@ -33,8 +33,8 @@ function testTransientSquareCavity(testCase)
 %% 0. Read input
 
 % Absolute tolerances
-absTol = 1e-15;
-absTol2 = 1e-15*1e2;
+absTol = 1e-12;
+absTol2 = 1e-12*1e2;
```
Comment (andreas-apostolatos) [inline unitTest/@testFEMThermalConductionAnalysis/testTransientSquareCavity.m]: Please revert the changes in the unit tests, I would like to fix them myself under Windows.

File: unitTest/@testFEMThermalConductionAnalysis/testTransientWallHeating.m [hunk 6]
```diff
@@ -34,8 +34,8 @@ function testTransientWallHeating(testCase)
 %% 0. Read input
 
 % Absolute tolerances
-absTol = 1e-15;
-absTol3 = 1e-15*1e3;
+absTol = 1e-12;
+absTol3 = 1e-12*1e3;
```
Comment (andreas-apostolatos) [inline unitTest/@testFEMThermalConductionAnalysis/testTransientWallHeating.m]: Please revert the changes in the unit tests, I would like to fix them myself under Windows.

File: unitTest/@testIGAMembraneAnalysis/testNonlinearMembraneAnalysis.m [hunk 7]
```diff
@@ -38,7 +38,7 @@ function testNonlinearMembraneAnalysis(testCase)
 %% 0. Read input
 
 % Define absolute tolerance
-absTol = 1e-15;
+absTol = 1e-12;
```
Comment (andreas-apostolatos) [inline unitTest/@testIGAMembraneAnalysis/testNonlinearMembraneAnalysis.m]: Please revert the changes in the unit tests, I would like to fix them myself under Windows.

File: unitTest/@testIGAMembraneAnalysis/testTransientNonlinearMembraneAnalysis.m [hunk 8]
```diff
@@ -46,7 +46,7 @@ function testTransientNonlinearMembraneAnalysis(testCase)
 %% 0. Read input
 
 % Define absolute tolerance
-absTol = 1e-14;
+absTol = 1e-12;
```
Comment (andreas-apostolatos) [inline unitTest/@testIGAMembraneAnalysis/testTransientNonlinearMembraneAnalysis.m]: Please revert the changes in the unit tests, I would like to fix them myself under Windows.

**Review by andreas-apostolatos — CHANGES_REQUESTED** [review]
Hi Marko,

thanks a lot for the excellent work! Can you please revert all changes you did in the unit tests? I know that some of them fail under windows due to very strict tolerances but I would like to take care of them myself as it is a very decisive part of the code.

Kind Regards,
Andreas

**Comment by MarkoLeskovar** [comment]
Dear Andreas,

This branch should be now finished and ready to merge.

**Review by andreas-apostolatos — APPROVED** [review]

**Pull request merged**